.github/workflows/dast.yml

```diff
@@ -1,8 +1,10 @@
 name: Dynamic Application Security Tests
 
-on:
+on:  
+  pull_request:
+    branches: [release*]
   push:
-    branches: [main]
+    branches: [main, release*]
```

Comment:
It would be better to pick either "on PR" or "on push". Running the workflow multiple times is redundant

Reply:
This rational is a bit complex, but I'm open to changing this... Let me explain

We want our DAST to run **before** we push to production, so on push to main makes perfect sense for the happy path. The complexity is when it comes to a hotfix, if the test only runs after code get to the release branch it could be released to production without the test passing, so that's why I added the on pull_request. However running on pull_request, doesn't test post merge, meaning their could be discrepancies that cause issues, so then we'd want to run it again post merge into a release branch, hence on push to release*.

I'm not sure of a better way to handle this, we might want to talk about it more with the diagram in front of us https://github.com/DTS-STN/secure-client-hub/wiki.

Reply:
I don't remember if Zap fails a PR if issues are found (perhaps just new issues). The process is a bit more manual in which it opens an [issue](https://github.com/DTS-STN/secure-client-hub/issues/10) with the project and describes what it finds for devs to review.

In the scenario of a hotfix or regular release we should be checking what Zap reports before clicking publish within TC. Another factor to consider is, if prod is down, we likely wont want to be waiting on ZAP to do a 5 minute scan while hotfixing.

Reply:
I absolutely agree with your last point; I'll remove the pull_request.